Review and fix

KristjanESPERANTO opened the pull request:
Thanks for this nice module! :slightly_smiling_face: 

I reviewed the code and have some change suggestions.

This will also fix #5.

KristjanESPERANTO committed:
Remove unused dependency "cheerio"

KristjanESPERANTO commented:
@bogenpirat Did you noticed this PR?

bogenpirat commented:
sorry for the late response - i did see this back in december, but it slipped my working memory. apologies for that. thanks for the review! i'm just not sure that i agree with some of your changes.

if i see this right, this is basically just everything run through a formatter along with some string interpolation improvements and removing the superfluous import/updating the readme for it, correct? please correct me if i overlooked something - the formatting changes forced me to go over every line and i may have missed something.

removing cheerio - good point, that should happen. i'm not a fan of the formatter changes. i presume that the string interpolation has some performance benefits in JS? it is code that runs so rarely, i don't think that it requires optimization.

then there's this gem:
![image](https://github.com/bogenpirat/MMM-WetterOnline/assets/1835496/c4930051-368f-4bcd-8e50-817d6423e8ef)
really?

KristjanESPERANTO committed:
Format markdown

KristjanESPERANTO committed:
Fix class name + format

KristjanESPERANTO committed:
Fix name, fix description, add keywords

KristjanESPERANTO commented:
> i'm just not sure that i agree with some of your changes.

No problem. I am ready to shrink the changes back to an acceptable level :slightly_smiling_face: 

> if i see this right, this is basically just everything run through a formatter along with some string interpolation improvements and removing the superfluous import/updating the readme for it, correct?

Besides that I profoundly changed the getUrl function.

> the formatting changes forced me to go over every line and i may have missed something.

I try to adapt the code to the formatting rules of MagicMirror, but that was obviously too much to put it here in one PR. Sorry. I have undone some of the formatting changes to make it a little easier to compare the changes. 

> i presume that the string interpolation has some performance benefits in JS?

I doubt that it has performance advantages. The approach is rather to write modern JavaScript code. It wouldn't be a problem if I had to undo this as well. Should I?

bogenpirat commented on MMM-WetterOnline.js:
```diff
@@ -15,77 +15,77 @@ Module.register("MMM-WetterOnline", {
-		if(notification  == "WETTERONLINE_RESULTS") {
+
+	socketNotificationReceived (notification, payload) {
+		if(notification === "WETTERONLINE_RESULTS") {
```
just out of educational curiosity - why is this necessary? i know about the type check in ===, but when would this be a pitfall with == for a string?

KristjanESPERANTO replied:
Failure to consider all edge cases can lead to unpredictable behaviour. For this reason, it is good practice to use strict operators by default, and to omit them only in exceptional cases.

Here is a nice table with the different outputs of == and ===: https://stackoverflow.com/a/359509

bogenpirat commented on node_helper.js:
```diff
@@ -30,60 +25,59 @@ module.exports = NodeHelper.create({
+	extractEvent (dailyData, hourlyData, body) {
 		// extract current temp
-		let currentTempMatch = body.match(/<div id="nowcast-card-temperature"[^>]*>.*?<div class="value">(\d+)<\/div>/ms);
+		let currentTempMatch = body.match(/<div id="nowcast-card-temperature"[\s\S]*?<div class="value">(-?\d+)<\/div>/ms);
```
`[\s\S]*` is equivalent to `.*`, no? i'd much prefer to keep my version in that part of the expression. at least then i know that the tag is closed and everything between it and the value is matched ungreedily.
i realize that i'm arguing about how to parse html with regex, but i'd prefer this part my way. the minus obviously has to be in here.

KristjanESPERANTO replied:
I don't remember why I adapted the first part 😔 I have now undone that.

And yes, the part with the minus is the important part. Otherwise it doesn't work with negative temperatures.

bogenpirat reviewed (changes requested):
looks pretty good, just one minor change i'd prefer. thanks again for the continued work!

KristjanESPERANTO committed:
Refactor + format
- Rework getUrl with fetch
- Uniform use of quotation marks
- Mark build-in packages
- Get rid of var
- Fix typo

bogenpirat commented:
LGTM, thanks again!
